lib/internal/per_context/domexception.js

```diff
@@ -50,17 +51,17 @@ const disusedNamesSet = new SafeSet()
 
 class DOMException {
```

Comment:
Since this now uses `ReflectConstruct`, it might be better to do the following so that the runtime knows not to unnecessarily allocate the `this` instance[^1], especially in the interpreter:
```suggestion
class DOMException extends Error {
  static {
    ObjectSetPrototypeOf(DOMException, FunctionPrototype);
  }

```

[^1]: https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ecmascript-function-objects-construct-argumentslist-newtarget

Reply:
I tried with JS console, but that doesn't seem to work.

```js
class Foo extends Error {}
Object.setPrototypeOf(Foo, Function.prototype);
new Foo();
```

<img width="433" src="https://github.com/user-attachments/assets/9f916b37-2877-478c-8e2e-d378290fc55f" />

Reply:
Using `extends` will not work because of the restriction that `super(...)` must be called.

```js
class Foo extends Error {
    constructor() {
        this.foo = "foo";
    }
}
Object.setPrototypeOf(Foo, Function.prototype);
new Foo();
```

<img width="437" src="https://github.com/user-attachments/assets/51303931-a53e-4ecc-afe8-777f4e989dd8" />

Reply:
I see, I understand.